homeassistant/components/jellyfin/media_player.py

```diff
@@ -0,0 +1,463 @@
+    @needs_data
+    def media_duration(self) -> int | None:
+        """Duration of current playing media in seconds."""
+        return int(self._data["NowPlayingItem"]["RunTimeTicks"] / 10000000)
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def media_content_id(self) -> str | None:
+        """Content ID of current playing media."""
+        return str(self._data["NowPlayingItem"]["Id"])
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def volume_level(self) -> float | None:
+        """Volume level of the media player (0..1)."""
+        return float(self._data["PlayState"]["VolumeLevel"] / 100)
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return bool(self._data)
+
+    def media_seek(self, position: int) -> None:
+        """Send seek command."""
+        self._client.jellyfin.remote_seek(self._id, int(position * 10000000))
+
+    def media_pause(self) -> None:
+        """Send pause command."""
+        self._client.jellyfin.remote_pause(self._id)
+
+    def media_play(self) -> None:
+        """Send play command."""
+        self._client.jellyfin.remote_unpause(self._id)
+
+    def media_play_pause(self) -> None:
+        """Send the PlayPause command to the session."""
+        self._client.jellyfin.remote_playpause(self._id)
+
+    def media_stop(self) -> None:
+        """Send stop command."""
+        self._client.jellyfin.remote_stop(self._id)
+
+    def play_media(
+        self, media_type: str, media_id: str, **kwargs: dict[str, Any]
+    ) -> None:
+        """Play a piece of media."""
+        self._client.jellyfin.remote_play_media(self._id, [media_id])
+
+    def set_volume_level(self, volume: float) -> None:
+        """Set volume level, range 0..1."""
+        self._client.jellyfin.remote_set_volume(self._id, int(volume * 100))
+
+    def mute_volume(self, mute: bool) -> None:
+        """Mute the volume."""
+        if mute:
+            self._client.jellyfin.remote_mute(self._id)
+        else:
+            self._client.jellyfin.remote_unmute(self._id)
+
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return a device description for device registry."""
+
+        if not self._data["Capabilities"]["SupportsPersistentIdentifier"]:
+            return DeviceInfo(
+                identifiers={(DOMAIN, "Jellyfin-clients")},
+                name="Jellyfin Client Service",
+                manufacturer="Jellyfin",
+                model="Jellyfin Clients",
+                entry_type=DeviceEntryType.SERVICE,
+            )
+
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._data["DeviceId"])},
+            manufacturer="Jellyfin",
+            model=self._data["Client"],
+            name=self.name,
+            sw_version=self._data["ApplicationVersion"],
+            via_device=(DOMAIN, self._data["ServerId"]),
+        )
+
+    async def async_browse_media(
```

Comment:
for future PR. moving theses methods to browse_media may be desirable for maintainability.

Reply:
I mainly considered merging this with the existing browse_media code, as they do the same jellyfin side, and are similar, on the HA end. Iirc. (it's been a month or so) that should work well with a constructor function passed in, instead of hardcoded types.

I just considered this PR large enough for now. so I kept it where I wrote it for now.